crypto/dso/dso_win32.c

```diff
@@ -458,10 +458,7 @@ static char *win32_name_converter(DSO *dso, const char *filename)
         ERR_raise(ERR_LIB_DSO, DSO_R_NAME_TRANSLATION_FAILED);
         return NULL;
     }
-    if (transform)
-        sprintf(translated, "%s.dll", filename);
-    else
-        sprintf(translated, "%s", filename);
+    BIO_snprintf(translated, len + 1, transform ? "%s.dll" : "%s", filename);
```

Comment:
This doesn't look right, since if `transform` is non-zero we will have allocated `len + 5` bytes.

How about something like

```
    len = strlen(filename);
    transform = ((strstr(filename, "/") == NULL) &&
                 (strstr(filename, "\\") == NULL) &&
                 (strstr(filename, ":") == NULL));
    /* If transform != 0, then we convert to %s.dll, else just dupe filename */
    len += (transform) ? 5 : 1;
    translated = OPENSSL_malloc(len);
    if (translated == NULL) {
        ERR_raise(ERR_LIB_DSO, DSO_R_NAME_TRANSLATION_FAILED);
        return NULL;
    }
    BIO_snprintf(translated, len, "%s%s", filename, transform ? ".dll" : "");
```

Reply:
Oh, right. Fixed essentially as you suggested.